multipool_price_fetcher/src/processor.rs

```diff
@@ -0,0 +1,88 @@
+use alloy::dyn_abi::DynSolValue;
+use alloy::primitives::{Address, U256};
+use alloy::providers::Provider;
+use anyhow::anyhow;
+use itertools::Itertools;
+use multipool_storage::hook::HookInitializer;
+use multipool_types::kafka::KafkaTopics;
+use rdkafka::producer::FutureProducer;
+use rdkafka::producer::FutureRecord;
+use std::collections::HashMap;
+use std::time::Duration;
+
+#[derive(Clone)]
+pub struct PriceFetcher<P: Provider + Clone + 'static> {
+    pub producer: FutureProducer,
+    pub delay: Duration,
+    pub multicall_chunk_size: usize,
+    pub multicall_address: Address,
+    pub rpc: P,
+}
+
+impl<P: Provider + Clone + 'static> PriceFetcher<P> {
+    pub async fn process_prices(
+        &self,
+        mp: Address,
+        assets: Vec<Address>,
+    ) -> anyhow::Result<HashMap<Address, U256>> {
+        let multipool_functions = multipool_types::Multipool::abi::functions();
+        let get_price_func = &multipool_functions.get("getPrice").unwrap()[0];
+
+        let mut prices = Vec::new();
+        let chunked_assets = assets
+            .iter()
```

Comment:
let's reuse here code from core/store/.../pg/price_fetcher.rs there is also a multicall to fetch prices